mfr/extensions/pdf/index.js

```diff
@@ -0,0 +1,228 @@
+
+/*PDFViewerApplication.setTitleUsingUrl = function(url) {
+
+    this.url = url;
+
+    var rel_canonical = document.createElement('link');
+    rel_canonical.href = url;
+    rel_canonical.rel = "canonical";
+    document.head.appendChild(rel_canonical);
+
+    this.baseUrl = url.split('#')[0];
+    let title = getPDFFileNameFromURL(url, '');
+    if (!title) {
+      try {
+        title = decodeURIComponent(getFilenameFromUrl(url)) || url;
+      } catch (ex) {
+        // decodeURIComponent may throw URIError,
+        // fall back to using the unprocessed url in that case
+        title = url;
+      }
+    }
+
+    this.setTitle(title);
+
+};*/
+
+import  'pdfjs-web/viewer.css';
```

Comment:
The rest of this file builds a version of the pdf viewer that can run in our iframe. This is also the latest version of pdf.js

Reply:
TODO add comments explaining what is happening and why